### PR TITLE
Simple support for Page Blobs

### PR DESCRIPTION
**Pull request by ahmetalpbalkan**
Here is simple support for page blobs. (contains a few other minor fixes I just noticed)

Does not contain anything related to page blob sequence numbers (implementation omitted as it's not a current need for docker-registry azure storage driver).

cc: @paulmey @snesha @sachin-jayant-joshi

### CODE REVIEW

File: clients/storage/blob.go [hunk 1]
```diff
@@ -621,6 +646,82 @@ func (b BlobStorageClient) GetBlockList(container, name string, blockType BlockL
...
+		return ErrNotCreated
+	}
+	return nil
+}
+
+// PutPage writes a range of pages to a page blob or clears the given range.
+// In case of 'clear' writes, given chunk is discarded. Ranges must be aligned
+// with 512-byte boundaries and chunk must be of size multiplies by 512.
+// See https://msdn.microsoft.com/en-us/library/ee691975.aspx
+func (b BlobStorageClient) PutPage(container, name string, startByte, endByte int64, writeType PageWriteType, chunk []byte) error {
```
Comment (paulmey) [inline clients/storage/blob.go]: PutPage*__s__* ?

Comment (paulmey) [inline clients/storage/blob.go]: or PutPageRange?

Comment (ahmetalpbalkan) [inline clients/storage/blob.go]: being consistent with the REST endpoint name.

**Comment by paulmey** [comment]
lgtm

**Pull request merged**